test/postgres/doc.go

```diff
@@ -0,0 +1,18 @@
+// Package postgres contains testing helpers for PostgreSQL databases.
+//
+// Most of the functions provide independent implementations of functionality
+// that the datastore/postgres package does internally in its exposed API.
+//
+// # Test Databases
+//
+// The [TestIndexerDB] and [TestMatcherDB] start database engines and create
+// suitable databases as needed. See the [test/integration] package for more
+// information on the specifics of the former.
+//
+// As part of the database setup, these functions can load SQL files. This
+// package searches both its embedded copy of the local "sql" directory and the
+// main test package's "testdata" director for these patterns:
+//   - all_*.psql
+//   - matcher_*.psql (for [TestMatcherDB])
+//   - indexer_*.psql (for [TestIndexerDB])
```

Comment:
While writing the documentation, I thought through that it would be annoying that, if/when we have multiple database backends, all the SQL has to be in some dialect that's database agnostic because I made the mistake of camping on the `.sql` extension for the helper files. So, I changed this to use a ".psql" extension and added the metadata to make sure GitHub marks it as SQL.

Reply:
SGTM